Implement Mailing

Feliix42 opened the pull request:
This PR closes #17 and adds mailing functionalities.

~Will remove the `[WIP]` label when it's ready for merge.~

Feliix42 committed:
Also bind server to local IPs

Feliix42 committed:
Implmement invite link mails (#17)

This commit implements mail functionalities for invite link creation. As
soon as an invite link is created, the system sends out a mail to the
respective user the link has been created for.
The commit makes the following changes:
- add `mail` module in core for all future mail functionalities
- makes struct `InviteLink` public to use it in the mail module
- add error class for mail-related issues
- Expand .env by adding mail configuration

Feliix42 committed:
Update dependencies

Feliix42 committed:
Add test mail function

V1ncNet commented on .env:
```diff
@@ -1 +1,7 @@
+MAIL_SENDER=payments@ascii.coffee
+MAIL_URL=mail.example.local
+MAIL_USER=username
+MAIL_PASSWORD=password
```
Consider new line at the end of files

Feliix42 replied:
Huh, this was unintentional. Surely a glitch in the editor config.

Feliix42 committed:
Implement skeleton for monthly report

Adds the route `/admin/cron/reports` and an associated skeleton function
that already checks for the correct Header data to be transmitted.
Expands the .env file by a CRON_SECRET and a name for the `From` field
of emails.

Feliix42 committed:
Add new field to user account for opt-in monthly reports

Feliix42 committed:
Merge branch 'master' into feature/mails

Feliix42 committed:
Implement monthly report switch in account and settings UI

Feliix42 committed:
[ui] only show digest option when mail address is set

Feliix42 committed:
Provide generic mail sending function, add report skeleton

Feliix42 committed:
Merge branch 'master' into feature/mails

Feliix42 committed:
Merge branch 'master' into feature/mails

Feliix42 committed:
[chore] cargo fmt

Feliix42 committed:
Remove minor version from lettre dependency

pixix4 commented on src/core/mail.rs:
```diff
@@ -0,0 +1,112 @@
+    let email = EmailBuilder::new()
+        // Addresses can be specified by the tuple (email, alias)
+        .to((
+            account.mail.as_ref().expect("No mail address provided"),
```
Please change `expect` to `ServiceError`

pixix4 commented on src/core/mail.rs:
```diff
@@ -0,0 +1,112 @@
+        .subject("[ascii pay] Test Mail")
+        .text("This is a test mail to verify that the miling system works.")
+        .build()
+        .unwrap();
```
Please change `unwrap` to `ServiceError`

Feliix42 committed:
Fix logical bug

Feliix42 committed:
address review results

Feliix42 committed:
Stupidity mistake

Feliix42 committed:
Remove unnecessary unwrap
